src/server/process_request.c

```diff
@@ -293,10 +293,23 @@ req_register_sched(conn_t *conn, struct batch_request *preq)
 		rc = PBSE_UNKSCHED;
 		goto rerr;
 	}
+
 	if (sched->sc_conn_addr != conn->cn_addr) {
-		rc = PBSE_BADHOST;
-		goto rerr;
+		if (pbs_conf.pbs_primary != NULL && pbs_conf.pbs_secondary != NULL) {
+			pbs_net_t addr = get_hostaddr(pbs_conf.pbs_primary);
```

Comment:
just curious; does the server can send a register_sched? why are you checking the connection against pbs_primary and secondary?

Reply:
The current design is that sched is sending the registration request to a server. At this condition, the server is verifying the requesting sched's host. Since the connection could be either from the primary or secondary host,  the validation is checking on both.

Reply:
This is fine. However, I found in my testing that after a failover, and when primary is back, the scheduler on the secondary server does not shut down - i think that must be shut down